### PR TITLE
pkg/kvstore/allocator: Standardize usage of logfields

### LINKED ISSUE #30512: Inconsistent usage of log fields in pkg/kvstore
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

Minor issue: I noticed that the code in `pkg/kvstore` seems to be using different log fields from the rest of the codebase: instead of using global ones from [pkg/logging/logfields/logfields.go](https://github.com/cilium/cilium/blob/d72e63048361bcbe85be260c390fdecbc4ad9ff7/pkg/logging/logfields/logfields.go), it has its own fields defined in [pkg/kvstore/allocator/logfields.go](https://github.com/cilium/cilium/blob/d72e63048361bcbe85be260c390fdecbc4ad9ff7/pkg/kvstore/allocator/logfields.go).

The issue with that is that some of them end up being inconsistent: for example the `identity` field is called "[identity](https://github.com/cilium/cilium/blob/d72e63048361bcbe85be260c390fdecbc4ad9ff7/pkg/logging/logfields/logfields.go#L62-L63)" in one place and "[id](https://github.com/cilium/cilium/blob/d72e63048361bcbe85be260c390fdecbc4ad9ff7/pkg/kvstore/allocator/logfields.go#L7)" in another. This means that if I need to find all logs concerning a particular identity, I have to use a filter like this:
image
(This example is from Datadog but it applies to all log monitoring tools)

It's a minor annoyance but it's easy to forget this and miss relevant logs. Maybe we can just remove all usage of custom fields from `pkg/kvstore/allocator/logfields.go`? 

### Cilium Version

`Daemon: 1.13.9 a65bfca8cc 2023-11-29T14:19:04-07:00 go version go1.20.11 linux/arm64`

### Kernel Version

`5.15.0-1050-aws`

### Kubernetes Version

`Server Version: version.Info{Major:"1", Minor:"26+", GitVersion:"v1.26.6", GitCommit:"219d3257e999424d258e7e3bbd3afc1979bba8e4", GitTreeState:"clean", BuildDate:"2023-07-05T10:36:16Z", GoVersion:"go1.19.10", Compiler:"gc", Platform:"linux/arm64"}`

### Sysdump

_No response_

### Relevant log output

_No response_

### Anything else?

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by antonipp**
As proposed in https://github.com/cilium/cilium/issues/30512, this removes usage of inconsistent log fields from `pkg/kvstore/allocator/logfields.go`.

Fixes: https://github.com/cilium/cilium/issues/30512

```release-note
pkg/kvstore/allocator: Standardize usage of logfields
```

### CODE REVIEW

File: pkg/kvstore/allocator/allocator.go [hunk 1]
```diff
@@ -189,7 +189,7 @@ func (k *kvstoreBackend) Get(ctx context.Context, key allocator.AllocatorKey) (i
 	// Only key1 should match
 	prefix := path.Join(k.valuePrefix, k.backend.Encode([]byte(key.GetKey())))
 	pairs, err := k.backend.ListPrefix(ctx, prefix)
-	kvstore.Trace("ListPrefix", err, logrus.Fields{fieldPrefix: prefix, "entries": len(pairs)})
+	kvstore.Trace("ListPrefix", err, logrus.Fields{logfields.Prefix: prefix, "entries": len(pairs)})
```
Comment (marseel) [inline pkg/kvstore/allocator/allocator.go]: non-blocking nit: could you add it too for `entries`?

**Review by marseel — APPROVED** [review]
Thanks :)

File: pkg/kvstore/allocator/allocator.go (hunk 1 above)
Comment (antonipp) [reply pkg/kvstore/allocator/allocator.go]: Good catch, fixed it!

Comment (marseel) [reply pkg/kvstore/allocator/allocator.go]: Thank you!

**Review by nathanjsweet — APPROVED** [review]

**Comment by aanm** [comment]
@antonipp the tests are failing. Can you rebase the PR against main which might have some of the tests failures already fixed? Thank you

**Comment by ldelossa** [comment]
@antonipp looks like this pull is suffering from a segfault:

https://github.com/cilium/cilium/actions/runs/8001559708/job/22181288312

If you don't believe this is attributed to this change you can try to rebase this PR and we can run tests again.

**Comment by joestringer** [comment]
@ldelossa I saw that recently on other PRs / main but I believe it has now been fixed.

@antonipp I'd suggest rebasing the PR against the tip of `main`, then we can re-run CI to confirm and get this merged.

**Comment by antonipp** [comment]
Thanks, just rebased it again!

**Pull request merged**